### PR TITLE
Create DNF cache with -y before upgrading

### PR DESCRIPTION
**Pull request by DemiMarie**
This is a prerequisite for https://github.com/QubesOS/qubes-core-agent-linux/pull/260

### CODE REVIEW

File: update/qubes-vm.sls [hunk 1]
```diff
@@ -20,8 +20,13 @@ update:
...
+{% elif grains['os_family'] == 'RedHat' %}
+    - require:
+      - cmd: dnf-makecache
 {% endif %}
 
 notify-updates:
   cmd.run:
     - name: /usr/lib/qubes/upgrades-status-notify
+    - onchanges:
+      - pkg: update
```
Comment (marmarek) [inline update/qubes-vm.sls]: Running it always is intentional, see [commit which introduced it](https://github.com/QubesOS/qubes-mgmt-salt-dom0-update/commit/4ae92f868f9561422ceaba89ab42658a0d7cb285#diff-a809564b9a460228971ed88df7b64afe3d7c5070c1354c39bd4bb3b64a1fb702)

File: update/qubes-vm.sls [hunk 2]
```diff
@@ -5,12 +5,12 @@ dsa-4371-update:
...
-{% if grains['os'] == 'Fedora' %}
-# workaround for https://bugzilla.redhat.com/1669247
-dnf list updates --refresh >/dev/null:
-  cmd.run
+{% elif grains['os_family'] == 'RedHat' %}
+dnf-makecache:
+  cmd.script:
+    - source: salt://update/dnf-makecache
+    - runas: root
+    - stateful: True
```
Comment (marmarek) [inline update/qubes-vm.sls:13]: Just curious - why stateful (which requires this extra script)? Contrary to `dsa-4371-update`, it isn't critical to block the other call if makecache fails. Just more friendly status, or something else?

**Review by marmarek — CHANGES_REQUESTED** [review]

File: update/qubes-vm.sls (hunk 2 above)
Comment (DemiMarie) [reply update/qubes-vm.sls:13]: Mostly just for the friendly status.  I got tired of seeing “changed = 2” when nothing had actually changed.

File: update/qubes-vm.sls (hunk 1 above)
Comment (DemiMarie) [reply update/qubes-vm.sls]: Good point.  I’ll revert this, and make it stateful in a future PR.

**Comment by DemiMarie** [comment]
That OpenQA failure looks to be because a package isn’t signed correctly.  What could cause that?

**Comment by marmarek** [comment]
No, it was missing `dnf -y makecache` after enabling metadata check. The test run with updated `upgrades-installed-check` (that passes `-y` to `dnf check-update`) is still running. Earlier runs with that change failed because of bug in https://github.com/QubesOS/qubes-core-agent-linux/pull/269, which I now excluded from the run.

**Pull request merged**